### DISCUSSION TITLE
Ktor coil network in coil 3 alpha-03

### DISCUSSION THREAD

**Discussion by mahramane** [coil-kt/coil#2088] [General]
Hi
in coil 3 alpha-03 networkClient and cacheStrategy is required and i don't know how can i set it.
i didn't find any description in coil documents.

        setSingletonImageLoaderFactory { context ->
            ImageLoader.Builder(context)
                .components {
                    add(NetworkFetcher.Factory(
                        networkClient = ?,
                        cacheStrategy = ?
                    ))
                }.diskCache {
                    DiskCache.Builder()
                        .directory(helper.getCacheDir().toPath())
                        .maxSizePercent(0.1)
                        .build()
                }
                .build()
        }
        
i want to add my ktor client and default cacheStrategy
please help me

**Comment by JunJaBoy** [comment]
if you are using coil with ktor, I think you can add `KtorNetworkFetcherFactory()`:


```kotlin
package coil3.network.ktor

/* ... */

@JvmName("factory")
fun KtorNetworkFetcherFactory() = NetworkFetcher.Factory(
    networkClient = { HttpClient().asNetworkClient() },
    cacheStrategy = { CacheStrategy() },
)

/* ... */
```

the code will look like:

```kotlin
    setSingletonImageLoaderFactory { context ->
        ImageLoader.Builder(context)
            .components {
                add(factory = KtorNetworkFetcherFactory()) // HERE
            }.diskCache {
                DiskCache.Builder()
                    .directory(helper.getCacheDir().toPath())
                    .maxSizePercent(0.1)
                    .build()
            }
            .build()
    }
```
